PrestaShop/ps_linklist
# Allow adding product categories

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Building on top of @gregor-doroschenko work in pr #66 . Most of the heavy lifting seems to be done already but there were still some rough edges.
| Type?         | new feature
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #2 
| How to test?  | Multiple self test scenarios below. All of the test cases start from clean prestashop 1.7.7.0 install.

- [x] Test viewing and editing old linkblocks. 
* go to BO design -> link widget
* click edit for "products" block (this was fixed in 0c10471)
* there should be a new block "categories" which allow to select from the shop categories
* add one category to the selected links
* view in FO that the link is shown
* check that the link href points to the right category
* check that the link title corresponds to the category description without html tags (this is for SEO)
* change shop setting to use clean urls
* check FO: the link is shown in its block, the link is correct (fixed in 1004534)

- [x] Test adding new linkblock
* go to BO design -> link widget
* click "new block" button
* there should be a new block "categories" available for selecting
* give the block a name "selected categories" in all languages (for some reason this requires all languages to be filled)
* select hook "displayFooter" (just for ease of testing :)
* choose at least one category to be shown (actually good test would also be to select none :)
* click save
* check FO: the link is shown in its block, the link is correct and the title is correct
* change shop setting to use clean urls
* check FO: the link is shown in its block, the link is correct and the title is correct

- [x] Test removing old linkblock
* in BO design -> link widget
* delete old "products" block
* delete the just created "selected categories" block

Again most of the praise for this goes to @gregor-doroschenko :rocket:  I just fixed the url generation when pretty urls is on that @sarahdib found in the original pr (I'm guessing that was the problem as the video didn't really show everything that is needed to debug or replicate the situation, but I did find the same situation while doing my own tests)
